pytorch_toolkit/nncf/nncf/README.md

```diff
@@ -0,0 +1,123 @@
+Each compression method receives its own hyperparameters that are organized as a dictionary and basically stored in JSON file that is deserialized when the training starts. Compression methods can be applied separately or together producing sparse, quantized or sparse + quantized models. For more information about the configuration files please refer to the samples.
+
+### Uniform quantization  w/ fine-tuning
+A uniform "fake" quantization method supports an arbitrary number of bits (>=2) which is used to represent weights and activations.
+####  Symmetric quantization
+The method performs differentiable sampling of the continuous signal (e.g. activations or weights) during forward pass, simulating inference with integer arithmetic. During the training we optimize the **scale** parameter that represents range (determined by min=-scale and max=scale) of the original signal using gradient descent.
+The sampling formula is the following:
+```math
+output = round(clamp(input * \frac{level\_high}{scale}, level\_low, level\_high)) * \frac{scale}{level\_high}
+```
+
+Where `level_low` and `level_high` represent the range of the discrete signal.
+ - For weights: $`level\_low=-2^{bits-1}+1, level\_high=2^{bits-1}-1`$
+ - For unsigned activations: $`level\_low=0,\; level\_high=2^{bits}-1`$
+ - For signed activations: $`level\_low=-2^{bits-1},\; level\_high=2^{bits}-1`$
+ 
+ You can use `num_init_steps` parameter from `initializer` group to initialize the values of `scale` and determine what activation should be signed or unsigned from the collected statistics during this number of steps. 
+
+**Algorithm parameters**:
+- `algorithm` - the name of the compression algorithm should be `quantization` in this case.
+- `bits` - number of bits,  `8` by default.
+- `num_init_steps` - a number of steps to calculate per-layer activations statistics that can be used for **scale** initialization.
+- `signed_activations` - if true then signed activations are allowed in the model (optional).
+- `signed_activation_scopes` - list of layers that require singed activations (optional).
+- `quantize_inputs` - if true than quantize input of the model (optional).
+- `ignored_scopes` - list of layers that should be excluded from the compression (optional).
+
+### Non-structured sparsity
+Sparsity algorithm zeros weights in convolutional and fully-connected layers in a non-structured way, so that zero values are randomly distributed inside the tensor. Most of the sparsity algorithms set less to zero the less important weights but the criteria of how they do it is different. The framework contains several implementations of sparsity methods.
+
+#### RB-Sparsity
+
+This section describes the RB-Sparsity (Regularization-Based) algorithm that is implemented in this framework. The method is based on $`L_0`$ regularization that encourages parameters of the model to become zero:
+
+```math
+    ||\theta||_0 = \sum_{i=0}^{|\theta|} \lbrack \theta_i = 0 \rbrack
```

Comment:
It's not works on Github.

Use https://www.codecogs.com/latex/eqneditor.php to render latex.
`![](https://latex.codecogs.com/png.latex?%7C%7C%5Ctheta%7C%7C_0%20%3D%20%5Csum_%7Bi%3D0%7D%5E%7B%7C%5Ctheta%7C%7D%20%5Clbrack%20%5Ctheta_i%20%3D%200%20%5Crbrack)`

![](https://latex.codecogs.com/png.latex?%7C%7C%5Ctheta%7C%7C_0%20%3D%20%5Csum_%7Bi%3D0%7D%5E%7B%7C%5Ctheta%7C%7D%20%5Clbrack%20%5Ctheta_i%20%3D%200%20%5Crbrack)

Reply:
Done

Reply:
Done